node/src/Pos/Explorer/Txp/Local.hs

```diff
@@ -95,17 +94,15 @@ eTxProcessTransaction itw = withBlkSemaphoreIgnoreTip $ eTxProcessTransactionNoL
 eTxProcessTransactionNoLock
     :: (ETxpLocalWorkMode ctx m)
     => (TxId, TxAux) -> m (Either ToilVerFailure ())
-eTxProcessTransactionNoLock itw@(txId, TxAux {taTx = UnsafeTx {..}}) = runExceptT $ do
-    tipBefore <- GS.getTip
+eTxProcessTransactionNoLock itw@(txId, txAux) = runExceptT $ do
+    let UnsafeTx {..} = taTx txAux
+    -- Note: this function is assumed to be run while we own
+    -- 'BlkSemaphore', so we can be sure GState won't change while we
+    -- are doing these operations.
```

Comment:
Another good example where a type would help. Types win over comments. :-)

(But this comment is useful too since it explains why it is correct)